### DISCUSSION TITLE
Hot reload GraphQL pipeline

### DISCUSSION THREAD

**Discussion by Sergiusmi** [ChilliCream/hotchocolate#5107] [Q&A]
Hello!

In my project GraphQL queries and mutations are dynamically generated based on the database context and compile as "in memory" assembly. 
There are classes:
- Models classes from DB
- MyDbContext - EF database context
- Query - dynamicaly generated class with code for all my resolvers
- Mutation - dynamicaly generated class with code for all my mutations
- Input and Output classes for my mutations

I can make GraphQL queries and mutations to existing database entities. I use hotchocolate methods (_AddQueryType_ and _AddMutationType_) for my generated classes using reflection. 

Here is a method to add dynamic context with graphql support:
```
public static IRequestExecutorBuilder AddInteractiveQuery<TModel>(
            this IServiceCollection owner, 
            IRequestExecutorBuilder queryExecuterBuilder,
            IModelCompiler<TModel> modelCompiler,
            IModelsGenerator<TModel> modelGenerator,
            StoreGeneratorOptions dbOptions, 
            GraphQLCodeGeneratorOptions grqlOptions) where TModel:class
        {
            var builder = new StoryAssemblyBuilder<TModel>(modelCompiler);
            var store = StoreContext<TModel>
                .Create(builder)
                .SetModelsGeneratorOptions(dbOptions)  
                .SetModelsGenerator(modelGenerator);

            var buildingResult = store.GetDataContextMetadata();
            var dbType = buildingResult.DbContextAssembly.GetType(store.GeneratorOptions.GetContextFullName());
            var queryType = buildingResult.DbContextAssembly.GetType(grqlOptions.GetQueryTypeFullName());
            var mutationType = buildingResult.DbContextAssembly.GetType(grqlOptions.GetMutationTypeFullName());

            owner.RegisterDBContext(dbType, store.GeneratorOptions.ConnectionString);

            queryExecuterBuilder
                .AddQueryType(queryType)
                .SetPagingOptions(new PagingOptions() { DefaultPageSize = 50, IncludeTotalCount = true })
                .AddOffsetPagingProvider<CustomOffsetPagingProvider>()
                .AddProjections()
                .AddFiltering()
                .AddSorting()
                .AddMutationType(mutationType)
                .AddDefaultTransactionScopeHandler();
            ;

            return queryExecuterBuilder;
        }
```
I want to regenerate my assembly by reloading "Query" class and "Mutation" class in runtime when DB structure changes (e.g. when a new field is added to a table in the DB, or when an existing field is deleted).

How can I do this?
Thanks.

**Comment by PascalSenn** [comment]
You can do this with type modules: https://chillicream.com/blog/2021/09/27/hot-chocolate-12#dynamic-schemas

The schema definition will be slightly different, as you have to operate on lower level APIs, but the type modules are what you are looking for.
They define a event, and every time `TypesChanged` is triggered, the types of this modules are reloaded
